### PR TITLE
CI: Push detached head to master

### PR DESCRIPTION
**Pull request by effigies**
```Bash
#!/bin/bash -eo pipefail
git config user.email "${GITHUB_EMAIL}"
git config user.name "CircleCI Deploy"
git add templateflow/conf/templateflow-skel.zip
git commit -m "rel(${CIRCLE_TAG}): Update S3 skeleton file [skip ci]"
git push origin master

[detached HEAD ce41662] rel(0.4.1rc2): Update S3 skeleton file [skip ci]
 1 file changed, 0 insertions(+), 0 deletions(-)
 rewrite templateflow/conf/templateflow-skel.zip (73%)
Warning: Permanently added the RSA host key for IP address '192.30.253.112' to the list of known hosts.

Total 0 (delta 0), reused 0 (delta 0)
To github.com:templateflow/python-client.git
   a0adcb7..0c0deb9  master -> master

```

### CODE REVIEW

**Comment by oesteban** [comment]
Didn't see this. Although most of the releases will be done off of master, I don't think this would fully improve the situation when releasing from a different branch. I'm testing that right now

**Comment by effigies** [comment]
I think our two approaches were equivalent.

**Comment by oesteban** [comment]
It is still merging the full branch into master when releasing from the branch

**Comment by effigies** [comment]
Right. Did yours not?

**Comment by oesteban** [comment]
Didn't check, assumed it wouldn't. Let me try again :(

**Comment by effigies** [comment]
In both cases, you're committing on top of the latest release, and pushing the new commit to master. It will try a fast-forward merge. If what you want is to check out master and not merge from the branch:

```Shell
git fetch origin
git checkout origin/master
git add ...
git commit ...
git push origin HEAD:master
```
